test/lib/ffi/ffi_jit_call.lua

```diff
@@ -1,154 +1,166 @@
-
-int8_t call_i_i8(int a) asm("call_i");
-uint8_t call_i_u8(int a) asm("call_i");
-int16_t call_i_i16(int a) asm("call_i");
-uint16_t call_i_u16(int a) asm("call_i");
-int call_i8_i(int8_t a) asm("call_i");
-int call_u8_i(uint8_t a) asm("call_i");
-int call_i16_i(int16_t a) asm("call_i");
-int call_u16_i(uint16_t a) asm("call_i");
-
-int __fastcall fastcall_void(void);
-int __fastcall fastcall_i(int a);
-int __fastcall fastcall_ii(int a, int b);
-int __fastcall fastcall_iii(int a, int b, int c);
-int64_t __fastcall fastcall_ji(int64_t a, int b);
-double __fastcall fastcall_dd(double a, double b);
-int __fastcall fastcall_pp_i(int *a, int *b);
-
-int __stdcall stdcall_i(int a);
-int __stdcall stdcall_ii(int a, int b);
-double __stdcall stdcall_dd(double a, double b);
-float __stdcall stdcall_ff(float a, float b);
-]]
-
-local lib = ffi.load("../clib/ctest")
-
-do
-  local x
-  for i=1,100 do
-    x = lib.call_10i(-42, 17, 12345, 9987, -100, 11, 51, 0x12345678, 338, -78901234)
+do --- ffi-jit-call
+  local ffi = require("ffi")
+  
+  ffi.cdef[[
+  int call_10i(int a, int b, int c, int d, int e, int f, int g, int h, int i, int j);
+  double call_10d(double a, double b, double c, double d, double e, double f, double g, double h, double i, double j);
+  float call_10f(float a, float b, float c, float d, float e, float f, float g, float h, float i, float j);
+  int64_t call_ij(int a, int64_t b);
+  bool call_b(int a) asm("call_i");
+  
+  int64_t call_max(double,double,double,double,double,double,double,double,double,double,double,double,double,double,double,double,double) asm("call_10d");
+  
+  int64_t call_10j_p(int a, int b, int c, int d, int e, int f, int g, int h, int i, const char *p) asm("call_10j");
+  
+  int8_t call_i_i8(int a) asm("call_i");
+  uint8_t call_i_u8(int a) asm("call_i");
+  int16_t call_i_i16(int a) asm("call_i");
+  uint16_t call_i_u16(int a) asm("call_i");
+  int call_i8_i(int8_t a) asm("call_i");
+  int call_u8_i(uint8_t a) asm("call_i");
+  int call_i16_i(int16_t a) asm("call_i");
+  int call_u16_i(uint16_t a) asm("call_i");
+  
+  int __fastcall fastcall_void(void);
+  int __fastcall fastcall_i(int a);
+  int __fastcall fastcall_ii(int a, int b);
+  int __fastcall fastcall_iii(int a, int b, int c);
+  int64_t __fastcall fastcall_ji(int64_t a, int b);
+  double __fastcall fastcall_dd(double a, double b);
+  int __fastcall fastcall_pp_i(int *a, int *b);
+  
+  int __stdcall stdcall_i(int a);
+  int __stdcall stdcall_ii(int a, int b);
+  double __stdcall stdcall_dd(double a, double b);
+  float __stdcall stdcall_ff(float a, float b);
+  ]]
+  
+  local lib = ""
+  if jit and jit.os then
+    if jit.os == "Linux" then
+      lib = ffi.load("clib/ctest")
+    elseif jit.os == "Windows" then
+        -- currently disabling this check on windows.
+        -- need to build libs first in order to perform checks.
+        return
+        -- lib = ffi.load("clib/ctest.dll")
+    end
```

Comment:
Wouldn't mingw report jit.os as Windows too?  If yes, then blindly skipping here doesn't make sense since you're supporting mingw above in the Makefile.  Perhaps the right thing to do here would be to attempt to load ctest.dll and if it fails, just skip the test instead of letting it fail.

Reply:
> Wouldn't mingw report jit.os as Windows too?

Ohh, never thought about that.

> Perhaps the right thing to do here would be to attempt to load ctest.dll and if it fails, just skip the test instead of letting it fail.

Done.